subsystem/undertow/src/main/java/org/wildfly/extension/camel/undertow/CamelUndertowHostService.java

```diff
@@ -403,6 +447,8 @@ public String toString() {
 
     class CamelUndertowEventListener implements UndertowEventListener {
 
+        private final ConcurrentMap<String, Boolean> existingContextPaths = new ConcurrentHashMap<>();
```

Comment:
Feels to me like we're abusing map usage a little here.

The map value is effectively worthless because its always set to `Boolean.TRUE`. Seems we're only ever interested in the key in order to do remove operations and work out if we need to add a new entry.

Can we not just use a List with `add()` / `remove()` / `contains()`?

Reply:
`Set` does not have an atomic counterpart of `putIfAbsent()`. does it? Synchronizing a Set would be costier, would it not?

Reply:
I'm thinking more about readability than saving a few CPU cycles.

Reply:
Ok, let's leave it as-is.